### DISCUSSION TITLE
Q: 'add mp4 support'

### DISCUSSION THREAD

**Discussion by atomGit** [complexlogic/rsgain#74] [General]
hello again!

what exactly does 'add mp4 support' mean? can rsgain now be run on mp4 files which contain both audio and video?

**Comment by complexlogic** [comment]
The short answer is yes.

The PR you referenced added the .mp4 file extension to the program's supported file types. While it is true that .mp4 files usually contain video and .m4a files are usually audio only, they are both MPEG-4 containers. This means they share a common metadata system, and the program's logic handles both identically. Any video stream in an MPEG-4 file will be ignored and the audio will be analyzed only.

**Comment by atomGit** [comment]
mp4 support is very interesting - looking at a few of the latest commits, it seems there was nothing more needed than adding support for the .mp4 extension, is that really so?

apparently mpv has supported RG since 2020 as i just found out [here](https://github.com/mpv-player/mpv/issues/3405) - do you know if rsgain RG tags are usable with mpv (when using the mpv option `--replaygain track`)?

mpv offers the following...

```
 --replaygain                     Choices: no track album (default: no)
 --replaygain-clip                Flag (default: no)
 --replaygain-fallback            Float (-200 to 60) (default: 0.000)
 --replaygain-preamp              Float (-150 to 150) (default: 0.000)
```

ps: sorry for opening an issue - sometimes i fail to check if discussions are enabled

**Reply by complexlogic** [reply]
According to the commit you linked, it uses the same ReplayGain tags that rsgain writes

**Comment by atomGit** [comment]
also... do you plan on adding support for any more video containers? i'm thinking of mkv mostly

**Reply by complexlogic** [reply]
Any container needs to be supported by both FFmpeg and TagLib. FFmpeg does support mkv, but TagLib currently does not. There is an [open PR](https://github.com/taglib/taglib/pull/967) to add mkv support TagLib, but it doesn't seem like any recent progress has been made. I likely won't add another dependency for mkv support, so it'll need to be implemented in TagLib for me to enable support in rsgain.

**Comment by atomGit** [comment]
thanks for the reply and your work on this project
